### PR TITLE
Added MNIST dataset utils (loading and splitting by class)

### PR DESCRIPTION
**Pull request by FishAres**
Some hopefully useful tools for MNIST. May generalize it to other datasets.

### CODE REVIEW

File: nupic/research/frameworks/continuous_learning/mnist_data_utils.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        train_dataset = ClasswiseDataset(
+            cachefilepath=data_dir,
+            basename="mnist_train_",
+            qualifiers=[class_],
+            transform=transform,
+        )
+
+        train_loader.append(
+            DataLoader(
+                train_dataset, batch_size=batch_size, shuffle=False, drop_last=True
```
Comment (subutai) [inline nupic/research/frameworks/continuous_learning/mnist_data_utils.py:66]: Don't you want shuffle=True?

File: nupic/research/frameworks/continuous_learning/mnist_data_utils.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        test_dataset = ClasswiseDataset(
+            cachefilepath=data_dir,
+            basename="mnist_test_",
+            qualifiers=[class_],
+            transform=transform,
+        )
+
+        test_loader.append(
+            DataLoader(
+                test_dataset, batch_size=batch_size, shuffle=False, drop_last=True
```
Comment (subutai) [inline nupic/research/frameworks/continuous_learning/mnist_data_utils.py:79]: I think batch_size can be much higher for testing, which would make validation go faster.
